### DISCUSSION TITLE
How to Externalise JavaScript from Data?

### DISCUSSION THREAD

**Discussion by MichaelCropper** [d3/d3#3651] [Q&A]
i.e. 

1. HelloWorld.html = Contains Data Elements
2. HelloWorld.js = Contains JavaScript magic of D3

Summary of problems I'm having: https://stackoverflow.com/questions/76076607/jsp-struggling-with-javascript-module-syntax

Seems something related to ES6 JavaScript module stuff (and/or my lack of understanding)

**Comment by Fil** [comment]
I'm sorry but this question doesn't actually seem related to D3. Maybe try to get a simpler example to work in your environment—something like console.log("Hello, world").

**Comment by MichaelCropper** [comment]
It may not in your opinion, but it is. 

I'm looking for a basic working example for something along the lines of this, but I can't find anything in the docs and anything I've tried throws a variety of errors with things not being defined properly i.e. "caught ReferenceError: BubbleChart is not defined" which is the error that is thrown in the below example;

```
<!DOCTYPE html>
<html>
    <head>
        <title>HelloWorld</title>
    </head>
    <body>
        <script type="module">
            import * as BubbleChart from "/JavaScript/HelloBubble.js";
        </script>
        <script type="module">

            import * as d3 from "https://cdn.jsdelivr.net/npm/d3@7/+esm";

            const svg = BubbleChart([["Hello", 10], ["World", 20]]);
            document.body.appendChild(svg);

        </script>    
    </body>
</html>

```
